sqlmesh/core/engine_adapter/mixins.py

```diff
@@ -337,3 +338,53 @@ def _build_clustered_by_exp(
         **kwargs: t.Any,
     ) -> t.Optional[exp.Cluster]:
         return exp.Cluster(expressions=[exp.column(col) for col in clustered_by])
+
+    def _parse_clustering_key(self, clustering_key: t.Optional[str]) -> t.List[exp.Expression]:
```

Comment:
I moved the cluster key changing logic out of the Snowflake adapter and into `ClusteredByMixin` so the BigQuery adapter could utilise it